### ISSUE TITLE
mysql-server has issues starting if an empty folder is mounted onto /var/lib/mysql for persistence

### ISSUE THREAD

**Issue by alan614** [laraedit/laraedit-docker#33]
I don't think this is a major issue as I'm guessing the docker image was intended to store the database solely within the container, but I was aiming of be able to persist the database across container destructions. To do this, I mounted a folder onto /var/lib/mysql of the container. This was one of the first things I did so the folder going into /var/lib/mysql was empty, I was assuming this would get populated which it did, however, I found that the mysql-server keeps stopping and starting:

> 2016-05-27 17:12:43,371 INFO spawned: 'mysql' with pid 6351
> laraedit_server_1 | 2016-05-27 17:12:44,372 INFO success: mysql entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
> laraedit_server_1 | 2016-05-27 17:12:45,905 INFO exited: mysql (exit status 0; expected)
> laraedit_server_1 | 2016-05-27 17:12:46,907 INFO spawned: 'mysql' with pid 6378
> laraedit_server_1 | 2016-05-27 17:12:47,909 INFO success: mysql entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
> laraedit_server_1 | 2016-05-27 17:12:49,582 INFO exited: mysql (exit status 0; expected)
> laraedit_server_1 | 2016-05-27 17:12:50,586 INFO spawned: 'mysql' with pid 6405

*I'm also using docker compose if that causes any issue

but it any case, I've resolved this by first starting up the container without mounting anything onto /var/lib/mysql so it starts with the data in the image, which works fine. I then stopped the mysql server (I had to mod the supervisor.conf) and copied /var/lib/mysql onto on of my other mount points. I stopped the container then destroyed it. I took the copy and mounted it onto the /var/lib/mysql of a new container and everything works fine.

So I wonder if it's possible for the mysql-server of the docker image to be able to initialize gracefully from an empty /var/lib/mysql folder. I've been able to come up with a workaround, but maybe it could be worth taking up in the future.

Thanks

**Comment by ibourgeois** [comment]
Thanks for the info. 

Mounting a new (empty) directory should only require the MySQL service to be restarted to correct it. When I get home, I will look into it! If this is the case then checking for an empty directory and restarting the MySQL service gracefully shouldn't be a very big task... in which case, I'll definitely take care of this for you! 

If something else is preventing it from working properly, it may take a little longer, but I will do what I can to get it fixed asap. You've given me quite a bit, but any further info you can send would be greatly appreciated.

Thanks again!

**Comment by alan614** [comment]
Hmmm, well, I pulled the image from docker hub instead of building it from the Dockerfile if that makes a difference.

Thanks!

**Comment by ibourgeois** [comment]
It shouldn't. The image is rebuilt regularly, anytime the Ubuntu image updated.

**Comment by sandrodz** [comment]
Is there an update regarding this? @iBourgeois 

Did you resolve this issue @alan614

**Comment by iBourgeois** [comment]
I was not able to replicate this, and received no further information. If you are having the same issue, please provide any relevant information and I will be happy to help!

**Comment by ibourgeois** [comment]
@sandrodz No, there has been no resolution to this issue yet. I was never able to replicate it to troubleshoot it. However, I recently got a new computer, so I will try again to replicate this problem and come up with a solution. 

In the meantime, if you have any relevant information, that would be great! Thanks.

**Comment by sandrodz** [comment]
its very simple, we want mysql to persist, not to have db imported or migrations run every container up.

so try to run this:

`docker run -p 80:80 -v /Users/sandro/Code/domain.ge:/var/www/html/app -v /Users/sandro/Code/domain.ge/mysql:/var/lib/mysql laraedit/laraedit`

So basically I'm trying to persist db files inside laravel project dir mysql. 

What happens next is continues exit and restart of mysql server.

What I did to fix this:

I started fresh container without volume /var/lib/mysql, cp-ed /var/lib/mysql to my laravel projects root dir, and then ran docker run with mysql volume mounted... it all worked.

**Comment by ibourgeois** [comment]
I think I get what is going on now. I'm saving this link here so I can look into when I'm on a computer that I have Docker installed on. http://blog.arungupta.me/docker-mysql-persistence/

**Comment by sandrodz** [comment]
maybe, but note that I'm running docker beta on OS X so I don't have VM, its native like on linux.

**Comment by ibourgeois** [comment]
Ah. I'm running OS X now too. So I will install docker Beta and try it out! Thanks!

**Comment by sandrodz** [comment]
nice :) I assume this mac is your new computer, in that case congrats.

Beta got some performance boosts lately so its pretty usable, I dropped boot2docker vm.

**Comment by alan614** [comment]
Hi, sorry. I didn't look into this issue any further. The only idea I can come up with is that a start up script needed to be in place that would recreate the mysql table should they go missing. It was something that I saw in the earlier versions of the shincoder homestead. Sorry I couldn't be of any more help

**Comment by sandrodz** [comment]
Workaround doesn't work in Centos + docker, but works in OS X (beta) + docker
